multitenant: add AdminMerge capability

ecwall opened the pull request:
Fixes #95138

AdminMerge is currently only called by the system tenant even though it is
named similarly to other Admin* functions so it does not need its own
capability for now.

This changes its required capability from noCapCheckNeeded to onlySystemTenant
to prevent secondary tenants from calling it.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/98517)
<!-- Reviewable:end -->

ecwall committed:
multitenant: alphabetize reqMethodToCap

Sort reqMethodToCap keys alphabetically for ease of reading.

Release note: None

ecwall committed:
multitenant: make AdminMerge onlySystemTenant

Fixes #95138

AdminMerge is currently only called by the system tenant even though it is
named similarly to other Admin* functions so it does not need its own
capability for now.

This changes its required capability from noCapCheckNeeded to onlySystemTenant
to prevent secondary tenants from calling it.

Release note: None

ecwall commented:
bors r=knz
